### ISSUE TITLE
warnings when compiling smp_svr with newlibc on 2.3.0

### ISSUE THREAD

**Issue by mfiumara** [zephyrproject-rtos/zephyr#27712]
**Describe the bug**
When compiling smp_svr sample, there's a bunch of warnings generated when compiling.

**To Reproduce**

`west build samples/subsys/mgmt/mcumgr/smp_svr -b nrf52dk_nrf52832 --pristine -- -DCONFIG_NEWLIB_LIBC=y`

**Expected behavior**
No warnings expected

**Impact**
Annoyance

**Logs and console output**
```
/home/mattia/azure/repos/bmsn-workspace/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c: In function 'img_mgmt_impl_erase_image_data':
/home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log_core.h:238:10: warning: format '%x' expects argument of type 'unsigned int', but argument 3 has type 'long unsigned int' [-Wformat=]
  238 |   printk("%c: " fmt "\n", z_log_minimal_level_to_char(_level), \
      |          ^~~~~~
/home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log_core.h:266:5: note: in expansion of macro 'Z_LOG_TO_PRINTK'
  266 |     Z_LOG_TO_PRINTK(_level, __VA_ARGS__);        \
      |     ^~~~~~~~~~~~~~~
/home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log_core.h:296:2: note: in expansion of macro '__LOG'
  296 |  __LOG(_level,           \
      |  ^~~~~
/home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log.h:40:25: note: in expansion of macro 'Z_LOG'
   40 | #define LOG_ERR(...)    Z_LOG(LOG_LEVEL_ERR, __VA_ARGS__)
      |                         ^~~~~
/home/mattia/azure/repos/bmsn-workspace/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:9: note: in expansion of macro 'LOG_ERR'
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |         ^~~~~~~
/home/mattia/azure/repos/bmsn-workspace/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:33: note: format string is defined here
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |                                ~^
      |                                 |
      |                                 unsigned int
      |                                %lx
In file included from /home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log.h:11,
                 from /home/mattia/azure/repos/bmsn-workspace/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:22:
/home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log_core.h:155:24: warning: format '%x' expects argument of type 'unsigned int', but argument 4 has type 'long unsigned int' [-Wformat=]
  155 | #define Z_LOG_STR(...) "%s: " GET_ARG1(__VA_ARGS__), __func__\
      |                        ^~~~~~
/home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log_core.h:188:30: note: in definition of macro '__LOG_INTERNAL'
  188 |    log_from_user(_src_level, __VA_ARGS__);   \
      |                              ^~~~~~~~~~~
/home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log_core.h:279:14: note: in expansion of macro 'Z_LOG_STR'
  279 |              Z_LOG_STR(__VA_ARGS__));\
      |              ^~~~~~~~~
/home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log_core.h:296:2: note: in expansion of macro '__LOG'
  296 |  __LOG(_level,           \
      |  ^~~~~
/home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log.h:40:25: note: in expansion of macro 'Z_LOG'
   40 | #define LOG_ERR(...)    Z_LOG(LOG_LEVEL_ERR, __VA_ARGS__)
      |                         ^~~~~
/home/mattia/azure/repos/bmsn-workspace/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:9: note: in expansion of macro 'LOG_ERR'
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |         ^~~~~~~
/home/mattia/azure/repos/bmsn-workspace/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:33: note: format string is defined here
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |                                ~^
      |                                 |
      |                                 unsigned int
      |                                %lx
In file included from /home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log.h:11,
                 from /home/mattia/azure/repos/bmsn-workspace/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:22:
/home/mattia/azure/repos/bmsn-workspace/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:17: warning: format '%x' expects argument of type 'unsigned int', but argument 3 has type 'long unsigned int' [-Wformat=]
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |                 ^~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                             |
      |                                                             long unsigned int
/home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log_core.h:188:30: note: in definition of macro '__LOG_INTERNAL'
  188 |    log_from_user(_src_level, __VA_ARGS__);   \
      |                              ^~~~~~~~~~~
/home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log_core.h:296:2: note: in expansion of macro '__LOG'
  296 |  __LOG(_level,           \
      |  ^~~~~
/home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log.h:40:25: note: in expansion of macro 'Z_LOG'
   40 | #define LOG_ERR(...)    Z_LOG(LOG_LEVEL_ERR, __VA_ARGS__)
      |                         ^~~~~
/home/mattia/azure/repos/bmsn-workspace/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:9: note: in expansion of macro 'LOG_ERR'
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |         ^~~~~~~
/home/mattia/azure/repos/bmsn-workspace/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:33: note: format string is defined here
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |                                ~^
      |                                 |
      |                                 unsigned int
      |                                %lx
In file included from /home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log.h:11,
                 from /home/mattia/azure/repos/bmsn-workspace/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:22:
/home/mattia/azure/repos/bmsn-workspace/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:17: warning: format '%x' expects argument of type 'unsigned int', but argument 2 has type 'long unsigned int' [-Wformat=]
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |                 ^~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                             |
      |                                                             long unsigned int
/home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log_core.h:291:27: note: in definition of macro '__LOG'
  291 |    log_printf_arg_checker(__VA_ARGS__);         \
      |                           ^~~~~~~~~~~
/home/mattia/azure/repos/bmsn-workspace/zephyr/include/logging/log.h:40:25: note: in expansion of macro 'Z_LOG'
   40 | #define LOG_ERR(...)    Z_LOG(LOG_LEVEL_ERR, __VA_ARGS__)
      |                         ^~~~~
/home/mattia/azure/repos/bmsn-workspace/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:9: note: in expansion of macro 'LOG_ERR'
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |         ^~~~~~~
/home/mattia/azure/repos/bmsn-workspace/modules/lib/mcumgr/cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:339:33: note: format string is defined here
  339 |         LOG_ERR("bad offset (0x%x)", fa->fa_off + num_bytes -1);
      |                                ~^
      |                                 |
      |                                 unsigned int
      |                                %lx
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: gnuarmemb
 - Zephyr tag 2.3.0

**Additional context**
N.A.

**Comment by MaureenHelm** [comment]
@mfiumara can you confirm this is now fixed?

**Comment by mfiumara** [comment]
Hi @MaureenHelm this looks to be resolved on master (tested on commit 94721b316f634c3a4c24f7a839ab87363248d0d2), so I'm closing this thanks for the response!
